lib/interactive_viewer.ml

```diff
@@ -69,11 +69,24 @@ let view (patches : Patch.t list) =
     | Some z -> Lwd.var z
     | None -> failwith "zipper_of_list: empty list"
   in
+  let curr_scroll_state = Lwd.var W.default_scroll_state in
+  let change_scroll_state action state =
+    let off_screen = action = `Action && state.W.position > state.W.bound in
+    let off_screen_content =
+      action = `Content && state.W.position > state.W.bound
+    in
+    if off_screen || off_screen_content then
```

Comment:
The exclusive `action = ...` checks (that cannot both be true at the same time) hints me at pattern matching:

```ocaml
  let off_screen =
    match action with
    | `Action -> ...
    | `Content -> ...
  in
```

I think that would make the code clearer, can you have a look ?

Reply:
Yeah! That would be a clearer implementation for these checks (because they cannot be both true simultaneously, as you said). I will get on making appropriate changes right away. Thanks for pointing it out!

Reply:
Sorry, I completely missed that the condition was the same in both cases !
The match always returns `true` in all branches so it can entirely be removed and replaced by `true`.
Then, `true && x` can be simplified to `x`.

Reply:
@Julow Yeah, I too did not notice this. Altho when I tried to do what you asked, I got the following error:
```
Error (warning 27 [unused-var-strict]): unused variable action.
```
I am stuck here, can you please help me?

Reply:
It means that the variable `action` is not used within the function and that you should make that clear. We usually do that by adding a `_` at the beginning of unused variables:

```
let change_scroll_state _action state =
```